src/components/flame-graph/Canvas.js

```diff
@@ -0,0 +1,239 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import * as React from 'react';
+import seedrandom from 'seedrandom';
+import withChartViewport from '../shared/chart/Viewport';
+import ChartCanvas from '../shared/chart/Canvas';
+import TextMeasurement from '../../utils/text-measurement';
+
+import type { Thread } from '../../types/profile';
+import type { CssPixels, UnitIntervalOfProfileRange } from '../../types/units';
+import type {
+  FlameGraphTiming,
+  FlameGraphDepth,
+  IndexIntoFlameGraphTiming,
+} from '../../profile-logic/flame-graph';
+import type { Action, ProfileSelection } from '../../types/actions';
+import type { CallNodeInfo } from '../../types/profile-derived';
+
+type Props = {
+  thread: Thread,
+  maxStackDepth: number,
+  containerWidth: CssPixels,
+  containerHeight: CssPixels,
+  viewportLeft: UnitIntervalOfProfileRange,
+  viewportRight: UnitIntervalOfProfileRange,
+  viewportTop: CssPixels,
+  viewportBottom: CssPixels,
+  flameGraphTiming: FlameGraphTiming,
+  callNodeInfo: CallNodeInfo,
+  stackFrameHeight: CssPixels,
+  updateProfileSelection: ProfileSelection => Action,
+  isDragging: boolean,
+};
+
+type HoveredStackTiming = {
+  depth: FlameGraphDepth,
+  flameGraphTimingIndex: IndexIntoFlameGraphTiming,
+};
+
+require('./Canvas.css');
+
+const ROW_HEIGHT = 16;
+const TEXT_OFFSET_START = 3;
+const TEXT_OFFSET_TOP = 11;
+
+function assignColor(depth, funcName) {
```

Comment:
I'd like to come up with a different color scheme from the pseudo-random flame colors. I think we should probably figure this out in a separate discussion issue, and not clutter up this PR. We could also land your work as-is, but not expose it to the tabs yet.

Reply:
Do you want me to leave the colors as is? Otherwise I'm fine just removing the coloring for now. That way don't introduce a dependency on `seedrandom` which we later might just remove again when we know what colors to use.

Reply:
Sounds good to remove the colors and the dependency.